arci/src/clients/partial_joint_trajectory_client.rs

```diff
@@ -103,3 +103,249 @@ where
+        };
+        let joint_names = vec![
+            String::from("part1"),
+            String::from("high"),
+            String::from("part2"),
+            String::from("low"),
+        ];
+        let correct = vec![1.0_f64, 3.0_f64];
+
+        let partial = PartialJointTrajectoryClient::new(joint_names.clone(), client.clone());
+        let current_pos = partial.current_joint_positions();
+        assert!(current_pos.is_ok());
+        let current_pos = current_pos.unwrap();
+
+        current_pos
+            .iter()
+            .zip(correct.iter())
+            .for_each(|(pos, correct)| assert_approx_eq!(*pos, *correct));
+    }
+
+    #[tokio::test]
+    async fn test_partial_send_pos() {
+        let client = DummyFull {
+            name: vec![String::from("part1"), String::from("high")],
+            pos: Arc::new(Mutex::new(vec![1.0_f64, 3.0_f64])),
+            last_trajectory: Arc::new(Mutex::new(Vec::new())),
+        };
+        let joint_names = vec![
+            String::from("part1"),
+            String::from("high"),
+            String::from("part2"),
+            String::from("low"),
+        ];
+        let next_pos = vec![2.2_f64, 0.5];
+        let duration = std::time::Duration::from_secs(5);
+
+        let partial = PartialJointTrajectoryClient::new(joint_names.clone(), client.clone());
+
+        let result = partial.send_joint_positions(next_pos.clone(), duration);
+        assert!(result.is_ok());
+        assert!(result.unwrap().await.is_ok());
+
+        let current_pos = partial.current_joint_positions().unwrap();
+        current_pos
+            .iter()
+            .zip(next_pos.iter())
+            .for_each(|(pos, correct)| assert_approx_eq!(*pos, *correct));
+    }
+
+    #[tokio::test]
+    async fn test_partial_trajectory() {
+        let client = DummyFull {
+            name: vec![
+                String::from("part1"),
+                String::from("high"),
+                String::from("part2"),
+                String::from("low"),
+            ],
+            pos: Arc::new(Mutex::new(vec![5.1_f64, 0.8, 2.4, 4.5])),
+            last_trajectory: Arc::new(Mutex::new(Vec::new())),
+        };
+        let joint_names = vec![
+            String::from("part1"),
+            String::from("high"),
+            String::from("part2"),
+            String::from("low"),
+        ];
+        let trajectories = vec![
+            TrajectoryPoint::new(
+                vec![1.0_f64, 3.0_f64, 2.2_f64, 4.0_f64],
+                std::time::Duration::from_secs(1),
+            ),
+            TrajectoryPoint::new(
+                vec![3.4_f64, 5.8_f64, 0.1_f64, 2.5_f64],
+                std::time::Duration::from_secs(2),
+            ),
+        ];
+        let correct = vec![3.4_f64, 5.8_f64, 0.1_f64, 2.5_f64];
+
+        let partial = PartialJointTrajectoryClient::new(joint_names.clone(), client.clone());
+        let result = partial.send_joint_trajectory(trajectories);
```

Comment:
if trajectories length is different from client name length,
I think ``fn send_joint_trajectory`` needs assert macro.

Now, part of trajectories are copied and others are zero.
length of the part of trajectories is same as client name length.